### ISSUE TITLE
renaming symbol from untitled python file does not change the symbol in the file

### ISSUE THREAD

**Issue by amunger** [microsoft/pylance-release#4335]
Type: <b>Bug</b>

1. create a new python file, do not save
2. reference a function from a saved local python file
3. rename the function from the untitled file's reference
🐛 symbol was renamed everywhere except the untitled file
seems to work correctly for untitled notebooks

Extension version: 2023.9.11291529
VS Code version: Code - Insiders 1.79.0-insider (f5d658848a7d39546e48b9393690b5ff18473a7b, 2023-05-04T05:21:45.650Z)
OS version: Windows_NT x64 10.0.22621
Modes:
Sandboxed: Yes

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz (8 x 1498)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.59GB (1.45GB free)|
|Process Argv|--crash-reporter-id 1202dc88-403d-4ebc-8393-af3551a8bd5b|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
vslsvsres303:30308271
pythontb:30258533
pythonptprofiler:30281269
vshan820:30294714
pythondataviewer:30285072
vscod805cf:30301675
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30404738
cppdebug:30492333
vsclangdf:30492506
c4g48928:30535728
dsvsc012:30540252
pynewext54:30618038
pylantcb52:30590116
pyind779:30611226
pythonsymbol12:30651887
6233i204:30725950
azdwalk:30721579
pythonms35:30671666
cai9j608:30680470
57b77579:30687741
pythonfmttext:30716741
pythoncmvfstr:30726892
fixhidewlkth:30724384
hidesbindicator:30724476
pythongtdpath:30726887

```

</details>

**Comment by debonte** [comment]
The correct behavior would be that nothing is renamed. We do allow renaming within untitled files and within/between cells in untitled notebooks, but rename shouldn't impact files outside of the untitled file.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2023.5.41, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023541-24-May-2023-PreRelease)
